tasmota/tasmota_xdrv_driver/xdrv_05_irremote_full.ino

```diff
@@ -221,37 +222,75 @@ void IrReceiveInit(void)
   irrecv->enableIRIn();                  // Start the receiver
 }
 
+namespace {
+  void addFloatToJson(JsonGeneratorObject& json, const char* key, float value) {
+    if (floorf(value) == value) {
+      json.add(key, (int32_t) floorf(value));       // integer
+    } else {
+      // TODO can do better here
+      json.addStrRaw(key, String(value, 1).c_str());   // non-integer, limit to only 1 sub-digit
+    }
```

Comment:
I suggest you use `ext_snprintf`. It already handles removing trailing digits if they are zeroes.

For example to have at most 2 digits (keep in mind that float is passed by pointer, not by value)
```
void addFloatToJson(JsonGeneratorObject& json, const char* key, float value) {
  char s[16];
  ext_snprintf(s, sizeof(s), PSTR("%-2_f"), &value);
  json.addStrRaw(key, s);
}
```

See https://tasmota.github.io/docs/Sensor-API/#extension-to-vsnprintf

Reply:
Thx, will refactor (TBH, I just copied the code from below as-is, w/o any second thoughts)

Reply:
Changed as suggested (added a separate value for `n/a` though as I found it non-intuitive to see `-100.0` when I was testing it. Please check if you're OK with it